application-task-manager-ui/src/main/resources/TaskManager/Code/TaskReportResultPage.xml

```diff
@@ -0,0 +1,178 @@
+    #if ($foreach.count &gt; 1)
+      #set ($userSql = "${userSql} OR ")
+    #end
+    #if ($isList)
+      #set ($userSql = "$!{userSql}${sqlName}.id.value like CONCAT('%', :${sqlName}${foreach.count}, '%')")
+    #else
+      #set ($userSql = "$!{userSql}${sqlName}.id.value like :${sqlName}${foreach.count}")
+    #end
+    #set ($discard = $whereParams.put("${sqlName}${foreach.count}", $user))
+  #end
+  #set ($whereSql = "$!{whereSql} AND ${sqlName}.id.id=obj.id AND ${sqlName}.id.name='${sqlName}' AND ($userSql)")
+#end
+{{/velocity}}
+{{velocity wiki="false"}}
+## &amp;&amp; "$!{request.outputSyntax}" == 'plain'
+#if($xcontext.action == 'get' )
+  $response.setContentType('application/json')
+  #set ($fromSql = ', StringProperty taskId')
+  #set ($whereParams = {})
+  ## Add Space filter.
+  #if("$!request.space" != '')
+    #foreach($space in $request.getParameterValues('space'))
+      #if ($foreach.count &gt; 1)
+        #set ($spaceSql = "${spaceSql} OR ")
+      #end
+      #set ($spaceSql = "$!{spaceSql}doc.space like :docspace${foreach.count} OR doc.space like CONCAT(:docspace${foreach.count}, '%')")
+      #set ($discard = $whereParams.put("docspace${foreach.count}", $space))
+    #end
+    #set ($whereSql = "AND (${spaceSql})")
+  #end
+  ## Add CreateDate filter.
+  #if("$!request.createDate" != '')
+    #set ($fromSql = "${fromSql}, DateProperty createDate")
+    #set ($whereSql = "$!{whereSql} AND createDate.id.id=obj.id AND createDate.id.name='createDate' AND createDate.id.value &gt;= :createDate")
+    #set ($discard = $whereParams.put('createDate', $datetool.toDate('yyyy/MM/dd', $request.createDate)))
+  #end
+  ## Add assignee filter.
+  #if("$!request.taskAssignee" != '')
+    #addUserFilter('assignee' 'taskAssignee' 1)
+  #end
+  ## Add assignee filter
+  #if("$!request.taskCreator" != '')
+    #addUserFilter('creator' 'taskCreator' 0)
+  #end
+  #gridresultwithfilter_buildQuery("$!request.classname" $request.collist.split(',') $fromSql "AND taskId.id.id=obj.id AND taskId.id.name='id' $!{whereSql}" $whereParams)
+  #set ($sql = "SELECT doc.fullName, taskId.value FROM XWikiDocument doc ${sql}")
+  #set($map = {})
+  #set($discard = $map.put('reqNo', $numbertool.toNumber($request.reqNo).intValue()))
+  #set($offset = $numbertool.toNumber($request.get('offset')).intValue())
+  ## Offset starts from 0 in velocity and 1 in javascript
+  #set($offset = $offset - 1)
+  #if(!$offset || $offset &lt; 0)
+    #set($offset = 0)
+  #end
+  #set($limit = $numbertool.toNumber($request.get('limit')).intValue())
+  #if(!$limit)
+    #set ($limit = 15)
+  #end
+  #set($query = $services.query.hql($sql))
+  #set ($query = $query.setLimit($limit).setOffset($offset).bindValues($sqlParams))
+  #set($items = $query.execute())
+  #set($discard = $map.put('totalrows', $items.size()))
+  #if ($limit &gt; 0)
+    #set($discard = $map.put('returnedrows', $mathtool.min($items.size(), $limit)))
+  #else
```

Comment:
You should have a default limit as otherwise this code could be used for attacking XWiki.